src/codegen/compile.ml

```diff
@@ -6253,6 +6288,19 @@ module MakeSerialization (Strm : Stream) = struct
             remember_failure get_val ^^
             get_val ^^ store_ptr
           )
+          )
+      | Prim Region ->
+         read_alias env (Prim Region) (fun get_arg_typ on_alloc ->
+          let (set_region, get_region) = new_local env "region" in
+          get_arg_typ ^^
```

Comment:
`get_arg_typ` feels misplaced here.  Given the name and other uses, my hunch is that it's only meaningful when the type we are reading is parameterized by a type argument, and `Region` is a primitive with no arguments.

Reply:
Might be, but I think arg_type might refer to the argument of the idl_alias constructor which is inserted to indicate the preservation of sharing. Worth investigating though. I only added the equality test as a sanity check but disabling it leads to other failures.

Reply:
> disabling it leads to other failures

I noticed those too.  The `check_region` routine fails for me, saying that `vec_pages` is bad.

Reply:
Woudln’t call it `get_arg_typ`, it’s more `get_idl_typ`. (If it were composite, we could extract the arg type from it, as in `Array Mut` above).